jwst/tests_nightly/general/nirspec/test_nirspec_steps_single.py

```diff
@@ -168,3 +171,107 @@ def test_nrs_fs_single_spec2(self):
                     'jw84600002001_02101_00001_nrs2_x1d_ref.fits')
                   ]
         self.compare_outputs(outputs)
+
+
+@pytest.mark.bigdata
+class TestNIRSpecMasterBackground_FS(BaseJWSTTest):
+    input_loc = 'nirspec'
+    ref_loc = ['test_masterbackground', 'nrs-fs', 'truth']
+    test_dir = ['test_masterbackground', 'nrs-fs']
+    rtol = 0.0001
+
+    def test_nirspec_masterbackground_fs_user1d(self):
+        """
+
+        Regression test of master background subtraction for NRS FS when a user 1-D spectrum is provided.
+
+        """
+        # input file has 2-D background image added to it
+        # produce a local copy
+        input_file = self.get_data(*self.test_dir,
+                                    'nrs1_sci_bkg_cal.fits',
+                                    docopy=True)
+        # user provide 1-D background was created from the 2-D background image
+        input_1dbkg_file = self.get_data(*self.test_dir,
+                                          'nrs1_bkg_x1d_user.fits',
+                                          docopy=True)
+
+        result = MasterBackgroundStep.call(input_file,
+                                           user_background=input_1dbkg_file,
+                                           save_results=True)
+
+        # _________________________________________________________________________
+        # Test 1 compare 4 FS 1D extracted spectra from sciene data with
+        # no background added to 4 FS 1D extracted spectra from the output
+        # from MasterBackground subtraction
+
+        # run 1-D extract on results from MasterBackground step
+        result_1d = Extract1dStep.call(result, save_results=True)
+
+        # run 1-D extract on original science data without background
+        input_sci_cal_file = self.get_data(*self.test_dir,
+                                            'nrs1_sci_cal.fits')
+        # find the 1D extraction of this file
+        # find Extract1dStep on sci data to use the same version of
+        # this rountine run on both files  rather than running it off line
+        # and having the 1-D extracted science file stored as input to step
+
+        sci_cal_1d = Extract1dStep.call(input_sci_cal_file, save_results=True)
```

Comment:
I don't think you need to specify save_results=True, since you're using the object that is returned.

Reply:
agree